pkg/utils/environment_variable.go

```diff
@@ -0,0 +1,58 @@
+package utils
+
+import (
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+)
+
+var defaultEnvVarValueOrDefault = envVarValueOrDefault
+
+type EnvVarValueOrDefaultFunc func(envVarName string, defaultValue string) string
+
+func IntFromEnvOrDefault(
+	envVarName string,
+	defaultValue string,
+	envVarValueOrDefaultFunc EnvVarValueOrDefaultFunc,
+) (int, error) {
+	if envVarValueOrDefaultFunc == nil {
+		envVarValueOrDefaultFunc = defaultEnvVarValueOrDefault
+	}
+
+	envVarValue := envVarValueOrDefault(envVarName, defaultValue)
+	envVarValueInt, err := strconv.Atoi(envVarValue)
+	if err != nil {
+		return -1, log.RecordedError(log.CAKC010E, envVarName, err.Error())
+	}
+
+	return envVarValueInt, nil
+}
+
+func DurationFromEnvOrDefault(
+	envVarName string,
+	defaultValue string,
+	envVarValueOrDefaultFunc EnvVarValueOrDefaultFunc,
+) (time.Duration, error) {
+	if envVarValueOrDefaultFunc == nil {
+		envVarValueOrDefaultFunc = defaultEnvVarValueOrDefault
+	}
+
+	envVarValue := envVarValueOrDefault(envVarName, defaultValue)
+	envVarValueDuration, err := time.ParseDuration(envVarValue)
```

Comment:
why is this needed when you can just get it via `6*time.seconds`?

Reply:
but what if the value is from the env and not the default value? in that case it will be a string.

Reply:
that's why i think it's more simple to have them both as strings. this will be the case for every env var regardless to the expected type. The value is always a string and the logic that converts it to the required type is in the util. it will create a uniformity throughout configurable values in this repo and in our other k8s projects.

Reply:
the 6 would be from the env and *time*seconds would be from the code

Reply:
yuck (sorry)

Reply:
also - the env should be `6m0s` to match 6 minutes. just `6` won't work.

Reply:
But no problem. You gave a good explanation

Reply:
would you prefer if the env var would be an int? e.g `CONJUR_TOKEN_TIMEOUT=6`? if so we would change the logic to `strconv.Atoi(envVarValue) * time.Minutes`

Reply:
the thing is that this will be a breaking change as today passing an int will not work. However, i believe that this variable was never used correctly and that setting `CONJUR_TOKEN_TIMEOUT=6` does not work also today. verifying now.

Reply:
yep, this won't work in the current product. i get a `ERROR: 2020/09/10 11:56:22 config.go:128: CAKC010E Failed to parse CONJUR_TOKEN_TIMEOUT. Reason: time: missing unit in duration 6` error.

@izgeri when this env var was created how did you intend to set it? with `6m0s`? do you know who defined this?
do you see a problem with changing this to an int input so it will be `CONJUR_TOKEN_TIMEOUT="6"` instead of `CONJUR_TOKEN_TIMEOUT="6m0s"`?

Reply:
once we get to a decision i will update the README with the valid value pattern.

Reply:
aha, i can see that this env var was used before as:
```
- name: CONJUR_TOKEN_TIMEOUT
+            value: "15s"
``` 
(so it in Slack) so i wouldn't change the UX to be of int to not break changes.

Reply:
@izgeri can you please review the commit i added with the readme change?

Reply:
I'm pretty sure that `CONJUR_TOKEN_TIMEOUT` was added for dev troubleshooting, though I see it is documented in the README

I think it is wise to set a standard and enforce it, we should just make sure that the standard (and the default, btw) is clear